pkg/minikube/bootstrapper/kubeadm/versions.go

```diff
@@ -158,10 +159,13 @@ func Supports(featureName string) bool {
 	return false
 }
 
-// ParseKubernetesVersion parses the kubernetes version
-func ParseKubernetesVersion(version string) (semver.Version, error) {
-	// Strip leading 'v' prefix from version for semver parsing
-	v, err := semver.Make(version[1:])
+// ParseKubernetesVersion validates the kubernetes version as legitimate
+func ParseKubernetesVersion(input string) (semver.Version, error) {
+	if !strings.HasPrefix(input, version.VersionPrefix) {
```

Comment:
I think this check can be removed: it isn't necessary to check for the leading 'v' here because it is stripped away immediately afterwards.  Just continue to use strings.TrimPrefix, and it will handle both cases perfectly.